bot.py

```diff
@@ -289,18 +339,26 @@ async def manage_reactions(payload, added: bool):
     else:
         await member.remove_roles(role)
 
+
 @bot.event
 async def on_member_join(member):
     botChannel = discord.utils.get(member.guild.channels, name='bot-stuff')
-    rulesChannel = discord.utils.get(member.guild.channels, name='rules-and-info')
-    await botChannel.send('Welcome to the server %s!\nPlease check out %s!\nIn order to view channels you need to add the relevant roles. Type !help for help, !serverroles for the roles you can add yourself to, !add "role1" "role2" to put yourself in that course.' %(member.mention, rulesChannel.mention))
+    rulesChannel = discord.utils.get(member.guild.channels,
+                                     name='rules-and-info')
+    await botChannel.send((
+        f'Welcome to the server {member.mention}!\nPlease check out {rulesChannel.mention}!\nIn order to view channels you need to add the relevant roles.\
+        Type !help for help, !serverroles for the roles you can add yourself to, !add "role1" "role2" to put yourself in that course.'
+    ))
+
 
 @bot.event
 async def on_raw_reaction_add(payload):
     await manage_reactions(payload, True)
 
+
 @bot.event
 async def on_raw_reaction_remove(payload):
     await manage_reactions(payload, False)
 
-bot.run(token.stringToken())
+
+bot.run(os.getenv('TOKEN'))
```

Comment:
I do have a question regarding using an environmental variable as opposed to an untracked .py file: What benefits does it provide?
My only reservations about pulling this change is the fact that I have a few different Tokens already set up in the format we currently have it, and am tentative to accept a change that means a modification to production without a benefit.

Reply:
I believe one of the main reasons is it is more standard to handle tokens in projects. I'm willing to do it for sake of being standard but I don't feel too strongly either way

Reply:
First, discord.py provides mechanisms for escaping certain Discord keywords (i.e. @everyone or @here) when you use User.mention. This feels more direct, especially because the bot is responding to a particular user. This also avoids (rare) username collisions with things like *Name* being converted to _Name_

Addressing the credentials change, I certainly understand that this is a rather pedantic change given the scope of this project. Using .env adds additional dependencies, and changes the workflow for the project. On the other hand, I would say env variables are more common for credentials and in general better practice (one argument could be that .env files are commonplace in .gitignore files). This change is just adopting a more standard approach to storing credentials, but I understand reservations to a change that doesn't really impact the project overall. 

As far as additional credentials are concerned, using the same syntax you can easily expose other credentials which makes adding secrets as simple as NAME=SECRET. 

Overall I know most of these changes are rather opinionated, so let me know what you think and I can make changes as necessary.

Reply:
I think for the moment everything but the token change is accepted so if you just, for the moment, comment it out and update this pull request I can easily accept it

Reply:
Removed the ".env" change. 
I would recommend looking into it, as it is the generally accepted method for storing and using credentials (language agnostic). It makes it easier (albeit only slightly) for someone to get an instance of the bot running: just adding a key=pair to .env instead of adding boilerplate functions in an ignored file. It can support multiple credentials also, which would be useful for adding functionality like YouTube streaming and the like.